src/common/dropdown/index.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    autoUpdate,
+    flip,
+    computePosition,
+    shift,
+    offset,
+    type ReferenceElement
+} from "@floating-ui/dom";
+
+interface DropdownUtilOptions {
+    reverse?: boolean;
+    offset?: number
+}
+
+export class DropdownUtil {
+    declare host: ReferenceElement;
+    declare overlay: HTMLElement;
+    declare cleanupFn: any;
+    declare options: DropdownUtilOptions;
+
+    constructor(host: HTMLElement, overlay: HTMLElement, options?: DropdownUtilOptions) {
+        this.host = host as ReferenceElement;
+        this.overlay = overlay as HTMLElement;
+        this.options = options ?? {};
+    }
+
+    show() {
+        this.cleanupFn = autoUpdate(
+            this.host,
+            this.overlay,
+            this.update.bind(this),
+        );
+    }
+
+    update() {
+        computePosition(this.host, this.overlay, {
+            placement: this.options.reverse ? "bottom-end" : "bottom-start",
+            middleware: [offset(this.options.offset ?? 4), flip(), shift()],
+        }).then(({ x, y }) => {
+            Object.assign(this.overlay.style, {
+                left: `${x}px`,
+                top: `${y}px`,
+            });
+        });
+    }
+
+    cleanup() {
+        this.cleanupFn?.();
+    }
+
+    hide() {
+        if (this.cleanup) this.cleanup();
+    }
```

Comment:
I don't understand why we have this `hide()` method in _addition_ to `cleanup()`, especially since to my knowledge that `if` statement will always be true. Why not just have `cleanup()`?

Reply:
It won't be true
Hide is there mostly there in order to distinguish between hiding the popup (where we have to run cleanup) or to cleanup. 
There was a case where this was getting called multiple times and caused a console error (even though the popup was closed) I believe this was for combobox when the popup goes away if there is no match and then comes back.